jlm/llvm/ir/domtree.cpp

```diff
@@ -7,6 +7,7 @@
 #include <jlm/llvm/ir/cfg.hpp>
 #include <jlm/llvm/ir/domtree.hpp>
 
+#include <functional>
```

Comment:
This is the only change in this file, so is it needed here?
I can't see that the include have been removed from any other header, which might have included it earlier.